### PR TITLE
Rewrite BETS payment confirmation API

### PR DESCRIPTION
**Pull request by esoergel**
@proteusvaccuum @orangejenny
Got some requirements changes for this - here are the updates
:tropical_fish:

### CODE REVIEW

File: custom/enikshay/integrations/bets/views.py [hunk 1]
```diff
@@ -99,54 +109,56 @@ def get_case(domain, case_id):
...
 
 
 @require_POST
 @csrf_exempt
 @login_or_digest_or_basic_or_apikey()
 @toggles.ENIKSHAY_API.required_decorator()
-def update_voucher(request, domain):
-    return _update_case_from_request(request, domain, VoucherUpdate)
-
+def payment_confirmation(request, domain):
```
Comment (esoergel) [inline custom/enikshay/integrations/bets/views.py]: The big change is now there's one endpoint handling multiple requests in bulk

File: custom/enikshay/integrations/bets/views.py [hunk 2]
```diff
@@ -30,67 +31,85 @@ def __init__(self, msg, status_code):
...
+        try:
+            return date_parser.parse(value)
+        except ValueError:
+            return super(FlexibleDateTimeProperty, self)._wrap(value)
 
-    case_type = CASE_TYPE_VOUCHER
+
+class PaymentUpdate(jsonobject.JsonObject):
+    id = jsonobject.StringProperty(required=True)
+    status = jsonobject.StringProperty(required=True, choices=['Success', 'Failure'])
```
Comment (orangejenny) [inline custom/enikshay/integrations/bets/views.py]: Could it be worth using constants here? The checks for `self.status == 'Success'` seem fragile.

File: custom/enikshay/integrations/bets/views.py [hunk 3]
```diff
@@ -99,56 +118,69 @@ def get_case(domain, case_id):
...
-            raise CaseNotFound()
-    except CaseNotFound:
+
+    if not isinstance(request_json.get('response', None), list):
+        raise ApiError(msg='Expected json of the form `{"response": []}`', status_code=400)
+
+    updates = []
+    for event_json in request_json['response']:
+        if event_json.get('event_type', None) not in ('Voucher', 'Incentive'):
+            raise ApiError(msg="Malformed JSON", status_code=400)
```
Comment (orangejenny) [inline custom/enikshay/integrations/bets/views.py]: If this is ever going to happen, would be nice for the error to be more specific.

**Review by orangejenny — APPROVED** [review]

**Comment by esoergel** [comment]
@orangejenny made a couple small changes from your feedback.

**Review by orangejenny — APPROVED** [review]
nice, thanks for updates

**Pull request merged**